shotgun_api3/shotgun.py

```diff
@@ -1473,10 +1497,13 @@ def update_project_last_accessed(self, project, user=None):
         record = self._call_rpc("update_project_last_accessed_by_current_user", params)
         result = self._parse_records(record)[0]
 
-
-    def _get_session_token(self):
-        """Hack to authenticate in order to download protected content
-        like Attachments
+    def get_session_token(self):
```

Comment:
This should still work - i just made the method public `get_session_token` rather than `_get_session_token` - so you can log in with username and password, then run `get_session_token()` to get a session token back. Subsequent calls to `get_session_token` will then return the same (cached) value over and over again - which should be fine given  that the session doesn't expire as long as it is being used. 

Does that sound right to you?